crates/hir-ty/src/method_resolution.rs

```diff
@@ -1113,6 +1113,25 @@ fn iterate_inherent_methods(
                 callback,
             )?;
         }
+
+        // try to find impls in the nearest ancestor block module
+        if let Some(impls) = db
+            .block_def_map(block_id)
+            .and_then(|map| map.parent())
+            .and_then(|module| module.containing_block())
+            .and_then(|block| db.inherent_impls_in_block(block))
+        {
+            impls_for_self_ty(
+                &impls,
+                self_ty,
+                table,
+                name,
+                receiver_ty,
+                receiver_adjustments.clone(),
+                module,
+                callback,
+            )?
+        }
```

Comment:
Instead of only handling the current and parent block def map, I think we should instead query the current block if it exists, and all its parent blocks